sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -419,7 +419,7 @@ case class CatalogTable(
     map ++= storage.toLinkedHashMap
     if (tracksPartitionsInCatalog) map.put("Partition Provider", "Catalog")
```

Comment:
`CatalogTable.toLinkedHashMap` was used for display only, not lookup. Shall we revisit all the key names here?